numba/targets/codegen.py

```diff
@@ -523,6 +536,58 @@ def resolve(self, engine):
             del self._unresolved[name]
 
 
+def _proxy(old):
+    @functools.wraps(old)
+    def wrapper(self, *args, **kwargs):
+        return old(self._ee, *args, **kwargs)
+    return wrapper
+
+
+class JitEngine(object):
+    """Wraps a ExecutionEngine to provide custom symbol tracking.
+    Since the symbol tracking is incomplete  (doesn't consider
+    loaded code object), we are not putting it in llvmlite.
+    """
+    def __init__(self, ee):
+        self._ee = ee
+        # Track symbol defined via codegen'd Module
+        # but not any cached object
+        self._defined_symbols = set()
```

Comment:
Should there be some symbol clobbering prevention mechanism?

Reply:
`llvm::ExecutionEngine` will catch duplicated symbols and we are not going to protect against that.  A proper duplicated symbol detection will need a more logic to check for the linkage (e.g. like `weak` linkage symbol can override).   This `_defined_symbols` set will be just enough to tell if a symbol exists and will not cause the `EE` symbol lookup to `exit(1)` when symbol-not-found.